Content.Shared/Preferences/HumanoidCharacterProfile.cs

```diff
@@ -597,6 +612,11 @@ public void EnsureValid(ICommonSession session, IDependencyCollection collection
                          .Where(prototypeManager.HasIndex)
                          .ToList();
 
+
+            var antagDisableTraits = AntagDisableTraitPreferences
+                .Where(id => prototypeManager.TryIndex(id, out var trait) && trait.AllowAntagDisable && traits.Contains(id))
+                .ToList();
```

Comment:
The list of valid trait prototypes may be changed by `GetValidTraits()`, so filtering for which traits are valid cannot be done here.